### ISSUE TITLE
How did you get started with Open Source?

### ISSUE THREAD

**Issue by ankurkaushal** [mdo/ama#153]
How did you get started with Open Source? As someone who loves working with OSS but wants to contribute as well, I want to know about your experience. Also, I made a small contribution to Bootstrap few months back, it was just couple of lines but it felt good. Now I want to contribute in much larger ways but big codebases intimidate me a bit, how do you suggest I go in that direction?

**Comment by mdo** [comment]
> How did you get started with Open Source?

My first exposure to it was by using GitHub to star and follow jQuery/WordPress projects that I used frequently. I did that for a couple years before actually open sourcing anything. My first open source project was actually Bootstrap and I had a lot of help on that part.

Jacob—the co-creator of Bootstrap—took care of structuring the project, readme, and Git stuff for us before we released it to the public. From there it was all the chaos of responding to issues, implementing features, etc. Kind of just threw ourselves into it :D.

> Also, I made a small contribution to Bootstrap few months back, it was just couple of lines but it felt good.

Hell yeah! <3

> Now I want to contribute in much larger ways but big codebases intimidate me a bit, how do you suggest I go in that direction?

Definitely a struggle we as maintainers face—how do we get folks new to the project involved easily? Come v4 of Bootstrap we should have some smaller tasks to get folks involved. Some other projects I know use labels like `beginner` or `15mins` to describe the effort level of a particular bug report. The best thing to do is if there's something in particular you'd like to do for a project, open an issue to describe your thoughts, ask questions, etc before opening a PR.
